server/upload.go

```diff
@@ -121,8 +91,6 @@ func uploadBlobChunked(ctx context.Context, requestURL *url.URL, layer *Layer, r
 
 			switch {
 			case resp.StatusCode == http.StatusUnauthorized:
-				ch <- errors.New("unauthorized")
```

Comment:
what happens to the progress writer on error?

Reply:
This was used to cleanup the goroutine. Since that goroutine doesn't exist, it no longer needed

Reply:
Added a separate error handler for max retries, which seemed to be missing from main